### ISSUE TITLE
Use Activity's defaultViewModelCreationExtras as default param in Fragment's activityViewModel extend function please.

### ISSUE THREAD

**Issue by MReP1** [InsertKoinIO/koin#1688]
When I use Koin's getActivityViewModel to get a ViewModel, I find that I can't get the Activity's Intent's extra in the SavedStateHandle. I found that Koin uses the Fragment's defaultViewModelCreationExtras by default, while Jetpack's activityViewModels uses the Activity's defaultViewModelCreationExtras.

```
class MyFragment: Fragment() {
    val viewModel by activityViewModel<MyViewModel>()
}
```

```
class MyViewModel(private val savedStateHandle: SavedStateHandle): ViewModel {
    init {
        // when I get from activity's intent, it exists. But null here.
        savedStateHandle.get<String>(key) // null
    }
}
```

When I use Jetpack's activityViewModels, it work.
And I found that Koin use fragment's `defaultViewModelCreationExtras` as default param in 3.5.1 version. as follows.

```
inline fun <reified T : ViewModel> Fragment.getActivityViewModel(
    qualifier: Qualifier? = null,
    noinline ownerProducer: () -> ViewModelStoreOwner = { requireActivity() },
    noinline extrasProducer: (() -> CreationExtras)? = null,
    noinline parameters: (() -> ParametersHolder)? = null,
): T {
    return resolveViewModel(
        T::class,
        ownerProducer().viewModelStore,
        extras = extrasProducer?.invoke() ?: this.defaultViewModelCreationExtras, // Koin use fragment's(this) but not activity's
        qualifier = qualifier,
        parameters = parameters,
        scope = getKoinScope()
    )
}
```

In Jetpack's activityViewModels, it use activity's `defaultViewModelCreationExtras`.as follows
```
@MainThread
public inline fun <reified VM : ViewModel> Fragment.activityViewModels(
    noinline extrasProducer: (() -> CreationExtras)? = null,
    noinline factoryProducer: (() -> Factory)? = null
): Lazy<VM> = createViewModelLazy(
    VM::class, { requireActivity().viewModelStore },
    { extrasProducer?.invoke() ?: requireActivity().defaultViewModelCreationExtras }, // Ktx use activity's.
    factoryProducer ?: { requireActivity().defaultViewModelProviderFactory }

)
```

And it issues that I cant migrate expediently, I have to  pass activity's defaultViewModelCreationExtras in Koin's activityViewModels. It look like 
```
class MyFragment: Fragment() {
    val viewModel by activityViewModel<MyViewModel>(
        extrasProducer = {
            requireActivity().defaultViewModelCreationExtras
        }
    )
}
```

this problem exists in Koin library and java compat library. I don't know if it was a design error or something other, I wish Koin could design it like the Jetpack library use activity defaultViewModelCreationExtras by default.

**Comment by arnaudgiuliani** [comment]
thanks for your feedback, interesting

**Comment by arnaudgiuliani** [comment]
Fixed in 3.5.2 👍
